src/Features/Lsif/Generator/Generator.cs

```diff
@@ -161,6 +183,7 @@ public async Task GenerateForProjectAsync(Project project, GeneratorOptions opti
             var documentVertex = new Graph.LsifDocument(new Uri(uri, UriKind.RelativeOrAbsolute), GetLanguageKind(semanticModel.Language), contentBase64Encoded, idFactory);
             lsifJsonWriter.Write(documentVertex);
             lsifJsonWriter.Write(new Event(Event.EventKind.Begin, documentVertex.GetId(), idFactory));
+            await GenerateSemanticTokensAsync(document, lsifJsonWriter, idFactory, documentVertex);
```

Comment:
i'd move this under GenerateDocumentFoldingRangesAsync.  it's in a very weird location right now (in the code that sets up the vertices/contains code).

Reply:
> i'd move this under GenerateDocumentFoldingRangesAsync

That feels like a strange home for it unless we rename that method. Semantic tokens isn't really conceptually related to folding ranges.

Reply:
Oh I see, you mean formatting-wise.

Reply:
to me, it's just a secondary feature.  just add at the end, just like we add GnerateDocumentFoldingRanges at teh end.